streaming/Dockerfile

```diff
@@ -0,0 +1,76 @@
+# syntax=docker/dockerfile:1.4
+
+# Please see https://docs.docker.com/engine/reference/builder for information about
+# the extended buildx capabilities used in this file.
+# Make sure multiarch TARGETPLATFORM is available for interpolation
+# See: https://docs.docker.com/build/building/multi-platform/
+ARG TARGETPLATFORM=${TARGETPLATFORM}
+ARG BUILDPLATFORM=${BUILDPLATFORM}
+
+# Node version to use in base image, change with [--build-arg NODE_MAJOR_VERSION="20"]
+ARG NODE_MAJOR_VERSION="20"
+# Debian image to use for base image, change with [--build-arg DEBIAN_VERSION="bookworm"]
+ARG DEBIAN_VERSION="bookworm"
+# Node image to use for base image based on combined variables (ex: 20-bookworm-slim)
+FROM docker.io/node:${NODE_MAJOR_VERSION}-${DEBIAN_VERSION}-slim as streaming
+
+# Timezone used by the Docker container and runtime, change with [--build-arg TZ=Europe/Berlin]
+ARG TZ="Etc/UTC"
+# Linux UID (user id) for the mastodon user, change with [--build-arg UID=1234]
+ARG UID="991"
+# Linux GID (group id) for the mastodon user, change with [--build-arg GID=1234]
+ARG GID="991"
+
+# Configure the IP to bind Mastodon to when serving traffic
+ENV BIND="0.0.0.0"
```

Comment:
We should make sure that we explicitly set the PORT, rather than just relying on the default in the code of 4000.

Reply:
@vmstan we also need `NODE_ENV=production` I think?

Reply:
That would only be for building a new image with it set to something though right? When executed through Docker Compose or Kubernetes couldn't you set that as a runtime environment variable still?

Reply:
It's set to expose port 4000 at the end, why would it also need to be here?

Reply:
nah, it's basically that whilst streaming defaults to `4000` at the moment, if that ever changed, by being explicit with setting `PORT=4000` you'd always get the `EXPOSE` to match, otherwise you'd have say `EXPOSE 4000` but the server running on `:5000` and everything would break. (it is a bit of paranoia but setting PORT explicitly ensures it'll never break in that way

Reply:
Basically node.js doesn't know about `EXPOSE`, it only knows about the environment variables.